lib/sqlalchemy/dialects/postgresql/base.py

```diff
@@ -1847,6 +1847,9 @@ def _on_metadata_drop(self, target, bind, checkfirst=False, **kw):
     sqltypes.Enum: ENUM,
     sqltypes.JSON.JSONPathType: _json.JSONPathType,
     sqltypes.JSON: _json.JSON,
+    sqltypes.DateTime: default._StrDateTime,
+    sqltypes.Date: default._StrDate,
+    sqltypes.Time: default._StrTime,
```

Comment:
I don't think we would want to use those subtipes these, since they transform to string using `str` that may not be understood by pg. I think it should define new subtype that render using the iso format

Reply:
Why are we using str for the default engine? AFAIK str() is Python specific, while ISO format is understood by most DBs. 

I mean I think it'd be better to change default.py to use isoformat().

Reply:
I checked SQLite's reference page and it supports ISO: https://sqlite.org/lang_datefunc.html

Reply:
> Why are we using str for the default engine

We are not. The default has no implementation. That _StrXX class with the implementation is used only by the string compiler, that is a mock compiler used by default when `str(<compiled>)` is called. It's not supposed to produce sql that is compatible with any particular db.

> I checked SQLite's reference page and it supports ISO: https://sqlite.org/lang_datefunc.html

Note that sqlite does not use that implementation, but has its own custom one:
https://github.com/sqlalchemy/sqlalchemy/blob/2df6eb140b4feb86d7cc99c0c976d1b22eb6e25b/lib/sqlalchemy/dialects/sqlite/base.py#L904-L910

I think that something similar should probably be added for the pg dialect

Reply:
Also see this comment regarding the reason for not having a default implementation https://github.com/sqlalchemy/sqlalchemy/issues/5052#issuecomment-569091948

Reply:
I don't quite understand what is the point of the string compiler. Who is using it and for what? Isn't this all used just for debugging purposes, when they DB is known?

Also, why can we not use ISO for "almost-all" dialects? We know that PG, MySQL and SQLite definitely supports it.

Reply:
>  Isn't this all used just for debugging purposes, when the dialect is known?

That is its purpose yes. it gets called when `str(stmt)` or `stmt.compile()` or  `stmt.compile(compile_kwargs={'literal_binds': True})` is used without specifying a dialect.

> Also, why can we not use ISO for "almost-all" dialects? We know that PG, MySQL and SQLite definitely supports it.

My objection is purely on using the defined _StrXX classes, that are defined only for the string compiler. We could have a mixin called someting like "IsoformatMixin" that can be used to define subtypes for each dialect.